app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java

```diff
@@ -1059,11 +1061,31 @@ private boolean isTabRestoredComplete() {
     public void showFindInPage() {
         final Session focusTab = sessionManager.getFocusSession();
         if (focusTab != null) {
+            View findInPageView = findInPage.getView();
+
+            PortraitStateModel portraitState = getPortraitStateModel();
+            if (portraitState != null) {
+                portraitState.request(findInPageView);
+                findInPage.setOnDismissListener(view -> {
+                    portraitState.cancelRequest(findInPageView);
+                    return null;
+                });
+            }
+
             findInPage.show(focusTab);
             TelemetryWrapper.findInPage(TelemetryWrapper.FIND_IN_PAGE.OPEN_BY_MENU);
         }
     }
 
+    private PortraitStateModel getPortraitStateModel() throws ClassCastException {
+        Activity activity = getActivity();
+        if (activity == null) {
+            return null;
+        }
+
+        return ((MainActivity) activity).getPortraitStateModel();
```

Comment:
I originally intended to trigger an exception in case the activity is not MainActivity, so incorrect usage can be located easily. Maybe I'll let it return null in production build, and still throw an exception in debug build.